src/UID2.Client/UID2Encryption.cs

```diff
@@ -513,16 +532,28 @@ private static IdentityScope DecodeIdentityScopeV3(byte value)
             return (IdentityScope)((value >> 4) & 1);
         }
 
-        private static IdentityType GetIdentityType(byte[] encryptedId)
+        private static bool ValidIdentityScopeAndType(byte value)
+        {
+            // 4th bit == 1 then EUID otherwise UID2
+            var identityScope = value >> 4;
+            // 2nd bit == 1 then Phone else Email
+            var identityType = (value & 0b1111) >> 2;
+
+
+            return (identityScope == (int)IdentityScope.UID2 || identityScope == (int)IdentityScope.EUID) && 
+                   (identityType == (int)IdentityType.Email || identityType == (int)IdentityType.Phone) &&
+                   // 1st and 0th bits are always 1 or always 0
+                   ((value & 0b11) == 0b11 || (value & 0b11) == 0) ;
+        }
+
+        private static IdentityType DecodeIdentityType(byte value)
         {
             // For specifics about the bitwise logic, check:
             // Confluence - UID2-79 UID2 Token v3/v4 and Raw UID2 format v3
             // In the base64-encoded version of encryptedId, the first character is always either A/B/E/F.
-            // After converting to binary and performing the AND operation against 1100,the result is always 0X00.
+            // After converting to binary and performing the AND operation against 0b100,the result is always 0X00.
             // So just bitshift right twice to get 000X, which results in either 0 or 1.
-            byte idType = encryptedId[0];
-            byte piiType = (byte)((idType & 0b_1100) >> 2);
-            return piiType == 0 ? IdentityType.Email : IdentityType.Phone;
+            return (IdentityType)((value & 0b100) >> 2);
```

Comment:
Can we change to `(IdentityType)((value >> 2) & 1)` to match `DecodeIdentityScopeV3`?

Reply:
If the bits are valid, this should work but sure

Reply:
Changed approach